llvm/lib/Target/AArch64/AArch64SVEInstrInfo.td

```diff
@@ -213,14 +213,31 @@ def AArch64fadd_p_contract : PatFrag<(ops node:$op1, node:$op2, node:$op3),
                                      (AArch64fadd_p node:$op1, node:$op2, node:$op3), [{
   return N->getFlags().hasAllowContract();
 }]>;
+def AArch64fadd_p_nostrictfp : PatFrag<(ops node:$op1, node:$op2, node:$op3),
+                                       (AArch64fadd_p node:$op1, node:$op2, node:$op3), [{
+    // We're allowed to introduce new trapping behaviour.
+    return !MF->getFunction().getAttributes().hasFnAttr(llvm::Attribute::StrictFP);
+}]>;
 def AArch64fadd_p_nsz : PatFrag<(ops node:$op1, node:$op2, node:$op3),
                                 (AArch64fadd_p node:$op1, node:$op2, node:$op3), [{
   return N->getFlags().hasNoSignedZeros();
 }]>;
+
+def AArch64fmul_p_nostrictfp : PatFrag<(ops node:$op1, node:$op2, node:$op3),
+                                       (AArch64fmul_p node:$op1, node:$op2, node:$op3), [{
+    // We're allowed to introduce new trapping behaviour.
+    return !MF->getFunction().getAttributes().hasFnAttr(llvm::Attribute::StrictFP);
```

Comment:
Instead of checking the function attributes like this, you probably want to add new nodes AArch64ISD::STRICT_FADD_PRED etc. for lowering ISD::STRICT_FADD etc.  You need a different node to handle the chain correctly anyway, I think.

(At the IR level, the intrinsics don't really interact correctly with strictfp because they're all marked IntrNoMem, but that's sort of separate.)

Reply:
I figured you might say that but this patch is more about improving the code generation for the non-strictfp case and so this code it just to maintain existing functionality for the -ftrapping-math case rather than to specifically improve support for strictfp.  Do you think this is acceptable rational or should I implement strictfp support properly first?

Reply:
I don't really want a pile of hacks to try to cover up the fact that the combination of strictfp and SVE intrinsics is broken.  If you don't care about that combination, please just drop this code.  If you do care, please fix it properly.

Reply:
OK, thanks for the feedback.  I'll look into SVE support for the STRICT ISD nodes.